Updated REST API user documentation to include missing calls

abrom opened the pull request:
Updated REST API user documentation to include missing calls:
 * createToken
 * getAvatar
 * register
 * resetAvatar

Updated REST API user documentation to include `username` as an optional query parameter:
* getPresence
* info
* setAvatar

Also fixed a few other minor issues with `setAvatar` and `update` calls

abrom commented on 6. Developer Guides/REST API/Users/createToken.md:
```diff
@@ -0,0 +1,36 @@
+## Change Log
+| Version | Description |
+| :--- | :--- |
+| 0.55.?? | Added |
```
N.B. This feature hasn't been released as yet so this will need to be updated when the information is available. Happy to hold up the PR until it has been released if that's easier :)

graywolf336 commented:
You want me to go ahead and merge the other pull request and wait for this one until the endpoint is for sure released to master branch?

abrom commented:
Hi @graywolf336, I've updated this to include the other changes RE username query parameter and fixed up the expected version release for the createToken function

reist commented on 6. Developer Guides/REST API/Users/resetAvatar.md:
```diff
@@ -0,0 +1,34 @@
+# User Reset Avatar
+| URL | Requires Auth | HTTP Method |
+| :--- | :--- | :--- |
+| `/api/v1/users.createToken` | `yes` | `POST` |
```
Shouldn't this be users.resetAvatar?

graywolf336 replied:
Good catch, mind changing this? I apologise for forgetting about this pull request

abrom replied:
Whoops.. good catch @reist 👍 

@graywolf336 fixed

graywolf336 commented:
Thanks again for this 👍
